internal/api/cluster.go

```diff
@@ -0,0 +1,140 @@
+package api
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-cloud/internal/model"
+)
+
+// outputJSON is a helper method to write the given data as JSON to the given writer.
+//
+// It only logs an error if one occurs, rather than returning, since there is no point in trying
+// to send a new status code back to the client once the body has started sending.
+func outputJSON(c *Context, w io.Writer, data interface{}) {
+	encoder := json.NewEncoder(w)
+	err := encoder.Encode(data)
+	if err != nil {
+		c.Logger.WithField("error", err).Error("failed to encode result")
+	}
+}
+
+// handleGetClusters responds to GET /api/clusters.
+func handleGetClusters(c *Context, w http.ResponseWriter, r *http.Request) {
+	pageStr := r.URL.Query().Get("page")
+	page, _ := strconv.Atoi(pageStr)
+
+	perPageStr := r.URL.Query().Get("per_page")
+	perPage := 100
+	if perPageStr != "" {
+		perPage, _ = strconv.Atoi(perPageStr)
```

Comment:
Do we want to log the error? It may help us uncover strange behavior in the short term at least. 1/5

Reply:
Indeed -- and I'm going to error out properly with `http.StatusBadRequest` here too. Thanks!